crates/sparrow-api/src/kaskada/v1alpha/schema_traits.rs

```diff
@@ -181,8 +181,11 @@ impl TryFrom<&arrow::datatypes::DataType> for DataType {
                     .try_collect()?;
                 Ok(DataType::new_struct(fields))
             }
-            arrow::datatypes::DataType::Map(s, _) => match s.data_type() {
-                arrow::datatypes::DataType::Struct(fields) => {
+            // Note: the `ordered` field may let us specialize the implementation
+            // to use binary search in the future.
+            arrow::datatypes::DataType::Map(s, _) => {
+                // [DataType::Map] is represented as a list of structs with two fields: `key` and `value`
+                if let arrow::datatypes::DataType::Struct(fields) = s.data_type() {
```

Comment:
Suggest alternating the order so the "error "is more clearly the "unhappy" path... but not a big deal.

```
let arrow::datatypes::DataType::Struct(fields) = s.data_type() else {
  return Err(ConversionError::new_unsupported(s.data_type().clone())
};

...
```